docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,240 @@ Note that:
+
+The MQTT message is as below:
+
+```json
+{
+   "configurations": [
+     {
+       "type": "<config type 1>",
+       "path": "</path/to/file/file 1>"
+     },
+     {
+       "type": "<config type 2>",
+       "path": "</path/to/file/file 2>"
+     },
+     ...
+   ]
+}
+```
+
+Example:
+```json
+{
+   "configurations": [
+     {
+       "type": "foo.conf",
+       "path": "/etc/child1/foo.conf"
+     },
+     {
+       "type": "bar.conf",
+       "path": "/etc/child1/bar.conf"
+     }
+   ]
+}
+```
+
+If path is specified, the `c8y_configuration_plugin` consumes/provides the configuration-file from/to the given local filesystem path. If the field `path` is not given, the `c8y_configuration_plugin` make use of the HTTP filetransfer feature of the `tedge_agent` to consume/provide the configuration-file (see [section below](#details-to-aspect-2-filetransfer-fromto-external-device) for more details about HTTP filetransfer). The first case is intended for local processes (running on the thin-edge device) that represent a child-device, and the latter case is intended for external devices.
+
+Whenever the `c8y_configuration_plugin` receivces that MQTT message it stores all contained information to a child-specific TOML file in `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml`. When the file already exists it will be replaced with the new content. The format of that TOML file is according to section [Configuration](#configuration) above. Each individual child's TOML file is stored in a subfolder, where the subfolder name is the `childid`.
+
+Example:
+
+```bash
+    $ tree /etc/tedge/c8y/
+    /etc/tedge/c8y/
+    ├── c8y-configuration-plugin.toml
+    ├── child1
+    │   └── c8y-configuration-plugin.toml
+    └── child2
+        └── c8y-configuration-plugin.toml    
+```
+
+The `c8y_configuration_plugin` adds per child-device the file `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml` implicitely to the child-device's config listed. So the list can always be configured from the cloud. The `type` for this self configuration file is `c8y-configuration-plugin`.
+
+Each time a child-device's TOML is modified (e.g. due to an incoming MQTT message, an incoming new configuration snapshot from the cloud, or a modification by a local process) the `c8y_configuration_plugin` takes care to define all necessary capabilities to the coresponding cloud's child-device twin. These are:
+  - declaring _supported operations_ for configuration management: `c8y_UploadConfigFile` and `c8y_DownloadConfigFile`
+  - declaring provided _configuration types_
+
+**Declaring 'supported operations'**
+
+To declare supported operations the `c8y_configuration_plugin` uses thin-edge's _Supported Operations API_. Therefore the `c8y_configuration_plugin` creates for each child-device two files under `/etc/tedge/operations/c8y/<childid>`.
```

Comment:
To make it explicit, we should mention that these operation files are also created by the plugin on receipt of that initial config list message.

Reply:
Now it is explicitely mentioned:

"[...] Each time a child-device's TOML is modified (e.g. due to an incoming MQTT message, an incoming new configuration snapshot from the cloud, or a modification by a local process) the c8y_configuration_plugin takes care to define all necessary capabilities to the coresponding cloud's child-device twin. [...]"